sql/sql_expression.h

```diff
@@ -495,6 +495,39 @@ struct GetAllColumnsOutput {
     std::shared_ptr<RowValueInfo> info;
 };
 
+/*****************************************************************************/
+/* COLUMN FILTER                                                             */
+/*****************************************************************************/
+
+/** Filter that rejects, accepts and or/modify a columns name.
+*/
+
+struct ColumnFilter {
+    typedef std::function<ColumnName (const ColumnName &)> Exec;
+   
+    ColumnFilter() 
+    {
+
+    }    
+
+    ColumnFilter(Exec exec)
+        : exec(std::move(exec))
+    {
+    }
+   
+    operator bool () const { return !!exec; }
+
+    Exec exec;
+
+    ColumnName operator () (const ColumnName & columnName) const
+    {
+        if (exec)
+            return exec(columnName);
+        else
+            return columnName;
```

Comment:
Consider creating a static member of type ColumnFilter named _identity_ to do just that.  This way the code is very explicit and we can prevent errors using a ColumnFilter that has wrongly not been initialized.